docs/javascript/typescript.md

```diff
@@ -0,0 +1,173 @@
+# TypeScript
+
+## Consuming WoltLab Suite’s Types
+
+To consume the types of WoltLab Suite you will need to install the [WoltLab/WCF](https://github.com/WoltLab/WCF) using npm using a git URL that refers to the appropriate branch.
+
+A full `package.json` that includes WoltLab Suite, TypeScript, eslint and Prettier could look like the following.
+
+```json
+{
+  "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^4.6.1",
+    "@typescript-eslint/parser": "^4.6.1",
+    "eslint": "^7.12.1",
+    "eslint-config-prettier": "^6.15.0",
+    "prettier": "^2.1.2",
+    "tslib": "^2.0.3",
+    "typescript": "^4.1.3"
+  },
+  "dependencies": {
+    "@woltlab/wcf": "https://github.com/WoltLab/WCF.git#master"
+  }
+}
+```
+
+After installing the types using npm you will also need to configure tsconfig.json to take the types into account.
```

Comment:
```suggestion
After installing the types using npm, you will also need to configure `tsconfig.json` to take the types into account.
```

May-be add the relevant `npm` commands (might apply in other places).

Reply:
I don't think we should explain how npm works, it's broadly known and we are not doing anything fancy. It's a simple `npm install`. I've only explained the use of `tsc` due to the use of `npx` in there, which is somewhat lesser known.